index.js

```diff
@@ -492,6 +498,31 @@ function getStyleFunction(glStyle, source, resolutions, onChange) {
           }
         }
 
+        if (type == 'Point' && 'circle-radius' in paint) {
+          ++stylesLength;
+          var cache_key = paint['circle-radius'] + '.' +
+            paint['circle-stroke-color'] + '.' +
+            paint['circle-color'];
+          style = iconImageCache[cache_key];
+          if(!style) {
```

Comment:
I just removed the if block.  style should *always* evaluate to true.  It will either come from the cache-object or be created.